test/models/driver_test.rb

```diff
@@ -47,35 +47,52 @@
       expect(new_driver.errors.messages[:name]).must_equal ["can't be blank"]
     end
 
-    it "must have a VIN number" do
+    it "must have a vin number" do
       # Arrange
       new_driver.vin = nil
 
       # Assert
       expect(new_driver.valid?).must_equal false
       expect(new_driver.errors.messages).must_include :vin
-      expect(new_driver.errors.messages[:vin]).must_equal ["can't be blank"]
+      expect(new_driver.errors.messages[:vin]).must_equal ["can't be blank", "is the wrong length (should be 17 characters)"]
     end
   end
 
   # Tests for methods you create should go here
   describe "custom methods" do
-    describe "average rating" do
-      # Your code here
-    end
+    describe "total_earniings" do
+      it "calculates the total amount earned by a driver on all trips" do
+        new_driver = Driver.create(name: "Waldo", vin: "ALWSS52P9NEYLVDE9")
+        passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+        trip_1 = Trip.create(driver_id: new_driver.id, passenger_id: passenger.id, date: Date.today, rating: 5, cost: 1234)
+        trip_2 = Trip.create(driver_id: new_driver.id, passenger_id: passenger.id, date: Date.today, rating: 3, cost: 6334)
 
-    describe "total earnings" do
-      # Your code here
-    end
+        assert_equal(57.9, new_driver.total_earnings)
+      end
 
-    describe "can go online" do
-      # Your code here
-    end
+      it "gives a total of nil if the driver has zero trips" do
+        no_trip_driver = Driver.create(name: "Mildred", vin: "TEST1234567891234")
 
-    describe "can go offline" do
-      # Your code here
+        assert_nil(no_trip_driver.total_earnings)
+      end
     end
 
-    # You may have additional methods to test
+    describe "ave_rating" do
+      it "calculates the average rating of a driver based on all trips" do
```

Comment:
I would also have a test with some complete trips and at least 1 incomplete trips (nil rating).